### PR TITLE
HW3. Just small part

### PR DESCRIPTION
**Pull request by Vladyslove**

### CODE REVIEW

File: src/test/java/homework/lesson3/DishTest.java [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+    public void mostLowCaloriesDishesStreamsTest() throws Exception {
+
+        Predicate<Dish> lowCalories = dish -> dish.getCalories() < 850;
+
+        List<String> lowCaloriesDishes = dishes.stream()
+                .filter(lowCalories)
+                .map(Dish::getName)
+                .collect(Collectors.toList());
+        for (String lowCaloriesDish : lowCaloriesDishes) {
+            System.out.println(lowCaloriesDish);
```
Comment (mfarsikov) [inline src/test/java/homework/lesson3/DishTest.java]: lowCaloriesDishes.forEach(System.out::println)

File: src/test/java/homework/lesson3/DishTest.java [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+
+    //1c- DONE
+    @Test
+    public void allDishesSortedByBioThenByAlphabet() throws Exception {
+
+        dishes = new ArrayList<>(dishes);
+
+        dishes.sort(new Comparator<Dish>() {
+            @Override
+            public int compare(Dish o2, Dish o1) {
```
Comment (mfarsikov) [inline src/test/java/homework/lesson3/DishTest.java]: 1. convert anonymous class to lambda
2. or even better: use `Comparator.comaring(Dish::isBio)`

File: src/test/java/homework/lesson3/DishTest.java [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+                return o1.getIsBio().compareTo(o2.getIsBio());
+            }
+        }.thenComparing(new Comparator<Dish>() {
+            @Override
+            public int compare(Dish o1, Dish o2) {
+                return o1.getName().compareTo(o2.getName());
+            }
+        }));
+        // the same is in the ONLY 1 line
+//        dishes.sort(Comparator.comparing(Dish::getIsBio).thenComparing(Dish::getName));
```
Comment (mfarsikov) [inline src/test/java/homework/lesson3/DishTest.java]: 👍

File: src/test/java/homework/lesson3/DishTest.java [hunk 4]
```diff
@@ -0,0 +1,154 @@
...
+        System.out.println(dishes);
```
Comment (mfarsikov) [inline src/test/java/homework/lesson3/DishTest.java]: it is required to return (or print) not `dish` but `dish.name`

File: src/test/java/homework/lesson3/DishTest.java [hunk 5]
```diff
@@ -0,0 +1,154 @@
...
+    @Test
+    public void pickedDishesByBioAndSortedByAlphabetStreamsTest() throws Exception {
+
+        Predicate<Dish> bio = dish -> dish.isBio;
+        List <String> bioDishes = dishes.stream()
+                .filter(bio)
+                .map(Dish::getName)
+                .collect(Collectors.toList());
+
+        bioDishes.sort(Comparator.naturalOrder());
```
Comment (mfarsikov) [inline src/test/java/homework/lesson3/DishTest.java]: sort can be done in stream pipeline (after .map())

File: src/test/java/homework/lesson3/DishTest.java [hunk 6]
```diff
@@ -0,0 +1,154 @@
...
+        Predicate<Dish> vegetables = dish -> dish.getType() == VEGETABLES;
+        OptionalDouble averagebeef = dishes.stream()
+                .filter(beef)
+                .mapToInt(Dish::getCalories)
+                .average();
+        OptionalDouble averagechicken = dishes.stream()
+                .filter(chicken)
+                .mapToInt(Dish::getCalories)
+                .average();
+        OptionalDouble averagevegetables = dishes.stream()
```
Comment (mfarsikov) [inline src/test/java/homework/lesson3/DishTest.java]: try to do this in single stream pipeline (result should be of type `Map<DishType, Double>`)

File: src/test/java/homework/lesson3/DishTest.java (hunk 1 above)
Comment (Vladyslove) [reply src/test/java/homework/lesson3/DishTest.java]: Agree! I don't know, why didn't I use forEach

File: src/test/java/homework/lesson3/DishTest.java (hunk 4 above)
Comment (Vladyslove) [reply src/test/java/homework/lesson3/DishTest.java]: Completely agree. Done.

**Comment by Vladyslove** [comment]
This pull request doesn't need now. There is another one.

**Pull request closed without merging**